OpenRA.Game/Map/Map.cs

```diff
@@ -760,9 +802,8 @@ public void FixOpenAreas(Ruleset rules)
 
 		public byte GetTerrainIndex(CPos cell)
 		{
-			var uv = cell.ToMPos(this);
-			var custom = CustomTerrain[uv];
-			return custom != byte.MaxValue ? custom : cachedTileSet.Value.GetTerrainIndex(MapTiles.Value[uv]);
+			var custom = CustomTerrain[cell];
+			return custom != byte.MaxValue ? custom : cachedTileSet.Value.GetTerrainIndex(MapTiles.Value[cell]);
```

Comment:
This is only faster because of the inline hack to special case the rectangular maps you've added to the CPos indexer in CellLayer.

This means for isometric maps you're going to be doing the CPos to MPos conversion still, and now you're going to be doing it twice. Considering the isometric conversion is vastly more expensive in comparison, you may want to avoid that.